lib/ddtrace/contrib/rails/core_extensions.rb

```diff
@@ -9,15 +9,33 @@ def patch_renderer
     end
 
     def patch_renderer_render_template
-      if defined?(::ActionView::Renderer)
-        ::ActionView::Renderer.class_eval do
+      if defined?(::ActionView::TemplateRenderer)
+        ::ActionView::TemplateRenderer.class_eval do
+          alias_method :render_without_datadog, :render
           alias_method :render_template_without_datadog, :render_template
-          def render_template(*args, &block)
-            ActiveSupport::Notifications.instrument('start_render_template.action_view')
-            render_template_without_datadog(*args, &block)
+
+          def render(context, options)
+            # create a tracing context and start the rendering span
+            @tracing_context = {}
+            ::ActiveSupport::Notifications.instrument('start_render_template.action_view', tracing_context: @tracing_context)
+            render_without_datadog(context, options)
+          rescue StandardError => e
+            # attach the exception to the tracing context if any
+            @tracing_context[:exception] = e
+            raise e
+          ensure
+            # ensure that the template `Span` is finished even during exceptions
+            ::ActiveSupport::Notifications.instrument('finish_render_template.action_view', tracing_context: @tracing_context)
+          end
+
+          def render_template(template, layout_name = nil, locals = nil)
```

Comment:
We patch this method because after the `render_template` we have the template and the layout computed. Computing these values in the `render`, means doing the same work twice (and we can even trigger an exception), so I think it's better instrument both methods.